contracts/strategies/BenqiStrategyAvaxV2.sol

```diff
@@ -0,0 +1,553 @@
+        }
+
+        amount = address(this).balance;
+
+        uint256 devFee = amount.mul(DEV_FEE_BIPS).div(BIPS_DIVISOR);
+        uint256 adminFee = amount.mul(ADMIN_FEE_BIPS).div(BIPS_DIVISOR);
+        uint256 reinvestFee = amount.mul(REINVEST_REWARD_BIPS).div(
+            BIPS_DIVISOR
+        );
+        WAVAX.deposit{value: devFee.add(adminFee).add(reinvestFee)}();
+        if (devFee > 0) {
+            _safeTransfer(address(rewardToken), devAddr, devFee);
+        }
+        if (adminFee > 0) {
+            _safeTransfer(address(rewardToken), owner(), adminFee);
+        }
+        if (reinvestFee > 0) {
+            _safeTransfer(address(rewardToken), msg.sender, reinvestFee);
+        }
+
+        _stakeDepositTokens(amount.sub(devFee).sub(adminFee).sub(reinvestFee));
+
+        emit Reinvest(totalDeposits(), totalSupply);
+    }
+
+    function _rollupDebt(uint256 principal, uint256 borrowed) internal {
+        (uint256 borrowLimit, uint256 borrowBips) = _getBorrowLimit();
+        uint256 supplied = principal;
+        uint256 lendTarget = principal.sub(borrowed).mul(leverageLevel).div(
+            leverageBips
+        );
+        uint256 totalBorrowed = borrowed;
+        while (supplied < lendTarget) {
+            uint256 toBorrowAmount = _getBorrowable(
+                supplied,
+                totalBorrowed,
+                borrowLimit,
+                borrowBips
+            );
+            if (supplied.add(toBorrowAmount) > lendTarget) {
+                toBorrowAmount = lendTarget.sub(supplied);
+            }
+            // safeguard needed because we can't mint below a certain threshold
+            if (toBorrowAmount < minMinting) {
+                break;
+            }
+            require(
+                tokenDelegator.borrow(toBorrowAmount) == 0,
+                "BenqiStrategyV1::borrowing failed"
+            );
+            tokenDelegator.mint{value: toBorrowAmount}();
+            supplied = tokenDelegator.balanceOfUnderlying(address(this));
+            totalBorrowed = totalBorrowed.add(toBorrowAmount);
+        }
+    }
+
+    function _getRedeemable(
+        uint256 balance,
+        uint256 borrowed,
+        uint256 borrowLimit,
+        uint256 bips
+    ) internal view returns (uint256) {
+        return
+            balance
+                .sub(borrowed.mul(bips).div(borrowLimit))
+                .mul(redeemLimitSafetyMargin)
+                .div(leverageBips);
+    }
+
+    function _getBorrowable(
+        uint256 balance,
+        uint256 borrowed,
+        uint256 borrowLimit,
+        uint256 bips
+    ) internal pure returns (uint256) {
+        return balance.mul(borrowLimit).div(bips).sub(borrowed);
+    }
+
+    function _getBorrowLimit() internal view returns (uint256, uint256) {
+        (, uint256 borrowLimit) = rewardController.markets(
+            address(tokenDelegator)
+        );
+        return (borrowLimit, 1e18);
+    }
+
+    function _unrollDebt() internal {
```

Comment:
I am suspecting this is true and can simplify the code:

`_unrollDebt() = _unrollDebtToFreeTarget(0)`